tests/SDK/Unit/Trace/NoopSpanBuilderTest.php

```diff
@@ -33,6 +26,41 @@ public function testSetParent(): void
         );
     }
 
+    public function testNoopCreatedSpanUsesProvidedContext(): void
```

Comment:
If we want to improve testability we should remove non-static global context interaction outside of the Context component.
I changed only `NoopSpanBuilder` to keep the scope of this PR small; as a follow-up we should update `Span`, `Tracer` and `TracerProvider` to accept a `ContextProvider` instead of interacting with the global context.

Reply:
Ok